src/SourceGenerators/Uno.UI.SourceGenerators/DependencyObject/DependencyObjectGenerator.cs

```diff
@@ -810,6 +810,15 @@ private void GenerateDependencyObjectImplementation(INamedTypeSymbol typeSymbol,
 
 				builder.AppendLineIndented("public void SetValue(DependencyProperty dp, object value) => __Store.SetValue(dp, value);");
 
+				if (_isUnoSolution)
+				{
+					builder.AppendLineIndented("internal void SetValue(DependencyProperty dp, bool value) => SetValue(dp, global::Uno.UI.Helpers.Boxes.Box(value));");
+
+					builder.AppendLineIndented("internal void SetValue(DependencyProperty dp, int value) => SetValue(dp, global::Uno.UI.Helpers.Boxes.Box(value));");
+
+					builder.AppendLineIndented("internal void SetValue(DependencyProperty dp, double value) => SetValue(dp, global::Uno.UI.Helpers.Boxes.Box(value));");
```

Comment:
These overloads are dangerous due to [implicit conversions](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/language-specification/conversions#1023-implicit-numeric-conversions):

- From `sbyte`, `short`, `ushort`, and `char` to `int`.
- From `sbyte`, `byte`, `short`, `ushort`, ~~`int`~~, `uint`, `long`, `ulong`, `char`, and `float` to `double`.

So for example, a float property calling SetValue will be improperly setting a double value rather than float, and later GetValue will throw invalid cast exception.

Reply:
I wrote an analyzer that catches such implicit conversions and report an error. But a better solution might be:

```
internal void SetValue<T>(DependencyProperty dp, T value)
{
    if (typeof(T) == typeof(bool)
    {
        SetValue(dp, Boxes.Box((bool)value);
    }
    // ... etc, for double and int.
    else
    {
        SetValue(dp, (object)value);
    }
}
```

Thoughts?

Reply:
what is the purpose of adding these typed overloads?

Reply:
When calling SetValue with a bool argument, it causes boxing to happen. Adding a non-object overload avoids that boxing, then we use a shared box.

Reply:
Using generics in that context may somehow expensive, and performing multiple type checks may be expensive as well. If we can statically detect those cases, it's probably best.